api/maven-api-core/src/main/java/org/apache/maven/api/SourceRoot.java

```diff
@@ -50,7 +49,7 @@ default Path directory() {
      * The default implementation returns an empty list, which means to apply a language-dependent pattern.
```

Comment:
Proposed new block of comment:

```
    /**
     * {@return the list of patterns for the files to include}.
     * The path separator is {@code /} on all platforms, including Windows.
     * The prefix before the {@code :} character, if present, is the syntax.
     * If no syntax is specified, the default is a Maven-specific variation
     * of the {@code "glob"} pattern.
     *
     * <p>The default implementation returns an empty list, which means to apply a language-dependent pattern.
     * For example, for the Java language, the default pattern is {@code "*.java"}.</p>
     */
```

Reply:
What if on windows we have a drive specified ? We need to differentiate:
```
C:/foo/**
```
So maybe rephrase as: `If the pattern does not start with {@code regex:} or {@code glob:}, the default variation of the {@code glob} pattern will be used.`

Reply:
Just to complete: "glob:" and "regex:" are the two syntaxes that all `FileSystem` shall support according NIO Javadoc, but a file system may support others syntaxes as well. It may be nice to avoid a formulation that restrict the possibilities to those two syntaxes.

Reply:
According Stackoverflow, Windows drives are restricted to 1 letter. So I suggest to replace _"If the pattern does not start with regex: or glob:"_ by _"The prefix before the : character, if present and longer than one character, is the syntax"_.